### PR TITLE
feat(explain): scope transcript and prompts to checkpoint boundaries

### PR DESCRIPTION
**Pull request by khaong**
## Summary
- When viewing a checkpoint with `entire explain -c {checkpoint}`, prompts and AI-generated summaries now show only content relevant to that specific checkpoint
- Uses the existing `TranscriptLinesAtStart` metadata field to slice the transcript at checkpoint boundaries
- Adds `SliceFromLine()` utility to the transcript package for byte-level slicing

## Test plan
- [x] Unit tests for `SliceFromLine()` covering edge cases (zero start, out of bounds, empty content, no trailing newline)
- [x] Integration tests for `scopeTranscriptForCheckpoint()` and `extractPromptsFromTranscript()`
- [x] End-to-end test `TestFormatCheckpointOutput_UsesScopedPrompts` verifying scoped prompts appear correctly
- [x] Manual test: run `entire explain -c {checkpoint}` on a checkpoint with multiple prior prompts and verify only relevant prompts are shown

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> User-visible `explain` output and summary generation now depend on correct transcript slicing; off-by-one or missing metadata could hide prompts or prevent summary generation. Changes are localized to CLI formatting/utilities but touch checkpoint listing behavior and parsing utilities.
> 
> **Overview**
> `entire explain --checkpoint` now *scopes prompts and generated summaries to the checkpoint’s portion of the transcript* by slicing from `TranscriptLinesAtStart`, instead of using the full session transcript.
> 
> Adds `transcript.SliceFromLine()` for byte-level JSONL slicing, updates branch checkpoint listing to derive the displayed session prompt from the scoped transcript, and extends tests to cover slicing edge cases plus scoped-prompt behavior and backwards-compatible fallback to stored prompts when no transcript exists.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 0e7a05f191612e56e9b5258fce8d314698df88d5. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: cmd/entire/cli/explain.go [hunk 1]
```diff
@@ -508,10 +553,10 @@ func formatCheckpointOutput(result *checkpoint.ReadCommittedResult, checkpointID
 
 		sb.WriteString("\n")
 
-		// Prompts section
+		// Prompts section (using scoped prompts, not full session)
 		sb.WriteString("Prompts:\n")
-		if result.Prompts != "" {
-			sb.WriteString(result.Prompts)
+		if len(scopedPrompts) > 0 {
+			sb.WriteString(strings.Join(scopedPrompts, "\n\n---\n\n"))
```
[the remark this answers is not in the document]
Comment (khaong) [reply cmd/entire/cli/explain.go:567]: this is intentional, show the full context of the conversation if `--full` is called

**Comment by khaong** [comment]
## Response to PR Review

**Re: Full mode transcript not scoped to checkpoint boundaries**

This is intentional behavior. The `--full` flag is meant to show the complete session transcript for context, while prompts are scoped to the checkpoint. This allows users to see exactly what happened during a checkpoint (scoped prompts) while still being able to reference the full conversation history if needed.

Added `TestFormatCheckpointOutput_FullShowsEntireTranscript` to document this as expected behavior.

---

**Re: Intent fallback ignores existing prompts when transcript missing**

Fixed in 265cf2d. Added backwards compatibility fallback: when a checkpoint has stored prompts but no transcript (older checkpoints), we now fall back to using `result.Prompts` instead of showing "(not generated)".

Added `TestFormatCheckpointOutput_FallsBackToStoredPrompts` to verify this behavior.
